src/kaleidoscope/device/key_indexes.h

```diff
@@ -17,21 +17,18 @@
 
 #pragma once
 
-typedef HARDWARE_IMPLEMENTATION::KeyAddr KeyAddr;
+typedef kaleidoscope::Device::KeyAddr KeyAddr;
```

Comment:
Rather include `KeyAddr.h`?

Reply:
Can't include `KeyAddr.h` here, because `KeyAddr.h` includes `KALEIDOSCOPE_HARDWARE_H`, which includes `key_indexes.h`. So we'd end up in an include loop. `KeyAddr.h` **must** include the hardware header, because it wants to have a resolvable `kaleidoscope::Device::KeyAddr`. So does `key_indexes.h`, but that's included after `kaleidoscope::Device` gets declared.

Yes, it is redundant, but we need the redundancy to avoid cycles. I'm up for exploring options about how to break the cycle, but in a follow-up PR. The same redundancy exists on master: [`KeyAddr.h`](https://github.com/keyboardio/Kaleidoscope/blob/10537e7f350fb6ecab4a2f423936ddd59f7cc214/src/kaleidoscope/KeyAddr.h#L23) & [`key_indexes.h`](https://github.com/keyboardio/Kaleidoscope/blob/10537e7f350fb6ecab4a2f423936ddd59f7cc214/src/kaleidoscope/hardware/key_indexes.h#L20). This isn't a newly introduced thing.

Reply:
Why does `KALEIDOSCOPE_HARDWARE_H` include key_indexes.h? Wouldn't it suffice for `Kaleidoscope.h` to include it. AFAIK, its included to provide convenience values to the user.

Reply:
You are correct, it does suffice, and it allows us to get rid of the redundancy in `key_indexes.h`. I'll update this PR in a bit then!